src/composables/useFetch.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ref,
+  watchEffect,
+  toValue,
+  ComputedRef,
+  type Ref,
+  type UnwrapRef,
+} from "vue";
+
+type UseFetchReturnType<T> = {
+  data: Ref<UnwrapRef<T> | null>;
+  error: Ref<UnwrapRef<Error> | null>;
```

Comment:
whats the purpose of unwrapping a ref and wrapping it again?

Reply:
I was typing it the wrong way and that's why I was getting this weird unwrapping ref types. Found a solution [here](https://github.com/orgs/vuejs/discussions/9564)